pkg/auth/agent_local_authenticator.go

```diff
@@ -66,6 +66,20 @@ func (a *AgentLocalAuthenticator) EnableOrgBasedFeatureGates() bool {
 }
 
 func (a *AgentLocalAuthenticator) AuthAgentAuth(token string) (interface{}, error) {
+	expiryDateTime, err := gencrypto.ParseExpirationFromToken(token)
+	// it is expected that agent installer sends the correct token
+	// and will contain the exp claim when it is required.
+	if err != nil && err.Error() != gencrypto.TokenMissingExpClaim {
+		a.log.WithError(err).Error("failed to parse token expiration time")
+	}
+	if expiryDateTime != nil {
+		expiryTime := time.Time(*expiryDateTime)
+		if expiryTime.Before(time.Now()) {
```

Comment:
IINW the [stored token](https://github.com/openshift/installer/pull/8717/files#diff-174c7f0a2bed95144baa2b38b752cb3a48aab7bc4380c16028faed7dd97924f9R91) is in UTC?

```suggestion
		if expiryTime.Before(time.Now().UTC()) {
```